mesonbuild/mdist.py

```diff
@@ -80,7 +79,19 @@ def is_git(src_root: str) -> bool:
     return quiet_git(['ls-files', '--error-unmatch', 'meson.build'], src_root)[0]
 
 def is_hg(src_root: str) -> bool:
-    return os.path.isdir(os.path.join(src_root, '.hg'))
+    '''
+    Checks if meson.build file at the root source directory is tracked by hg.
+    '''
+    try:
+        out = subprocess.check_output(
+            ['hg', 'status', 'meson.build', '--all'],
+            cwd=src_root, text=True, encoding='utf-8',
+        )
```

Comment:
`Popen_safe()` is generally used in the Meson codebase instead of using the `subprocess` module functions directly. This would also take care of dealing with systems where the default encoding is not `utf-8`. `encoding='utf-8` is still wrong here.